src/harness/testcases/WINNF_FT_S_FDB_testcase.py

```diff
@@ -0,0 +1,1014 @@
+#    Copyright 2018 SAS Project Authors. All Rights Reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+
+import json
+import os
+import sas
+import sas_testcase
+from util import configurable_testcase, writeConfig, loadConfig
+
+
+class FederalGovernmentDatabaseUpdateTestcase(sas_testcase.SasTestCase):
+  """SAS federal government database update test cases"""
+
+  def setUp(self):
+    self._sas, self._sas_admin = sas.GetTestingSas()
+    self._sas_admin.Reset()
+
+  def tearDown(self):
+    pass
+
+  def generate_FDB_1_default_config(self, filename):
```

Comment:
At the beginning of each test case, there is the following text:

> For clarity, this test is written with a single CBSD and protected entity. The test code shall allow for parallel execution of non-interacting tests in order to reduce total test execution time.

I think that was missed during the implementation. Please let me know if you have any questions.

Reply:
Does this mean that we shall allow multiple instances of e.g. FDB.1 to run concurrently or we shall allow concurrent parallel instances of different FDB test cases, e.g. one FDB.1, one FDB.2 ....?

Reply:
It means that you need to take in N CBSDs and expect that there will be at least N exclusion zones in the fake DB. So for example ITS may configure one exclusion zone in NY and another in CA, with one CBSD near each. The zones are far enough away from each other that the zones and CBSDs will not interfere with one another, but the SAS is processing all at the same time and the test harness is checking the results for all at the same time.

The goal was to save some time by allowing ITS to test N scenarios but only calling CPAS once; the alternative (run the test case N times) would require running CPAS N times, which unnecessarily takes more time.

So really the code is not very different at this stage, you just need to take in N registration requests instead of 1, and N*m grant requests instead of only m. The DB part is the same either way (but anyway that's not part of this PR). Does that make sense?

Reply:
Modified   all the test cases, Introduced Multiple CBSDs and Grants (N*m) in required   test cases.